sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbird/integration/service/SunbirdAuthenticationService.java

```diff
@@ -0,0 +1,145 @@
+import io.mosip.esignet.api.exception.KycExchangeException;
+import io.mosip.esignet.api.exception.SendOtpException;
+import io.mosip.esignet.api.spi.Authenticator;
+import io.mosip.esignet.api.util.ErrorConstants;
+import io.mosip.esignet.sunbird.integration.dto.KycExchangeRequestDto;
+import io.mosip.esignet.sunbird.integration.dto.KycExchangeResponseDto;
+import io.mosip.kernel.core.http.ResponseWrapper;
+import io.mosip.kernel.core.util.StringUtils;
+import io.mosip.kernel.keymanagerservice.dto.AllCertificatesDataResponseDto;
+import io.mosip.kernel.keymanagerservice.dto.CertificateDataResponseDto;
+import io.mosip.kernel.keymanagerservice.service.KeymanagerService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.annotation.PostConstruct;
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.*;
+
+
+@ConditionalOnProperty(value = "mosip.esignet.integration.authenticator", havingValue = "SunbirdAuthenticationService")
+@Component
+@Slf4j
+public class SunbirdAuthenticationService implements Authenticator {
+
+    private static final String APPLICATION_ID = "MOCK_AUTHENTICATION_SERVICE";
+
+    @Value("${mosip.esignet.mock.authenticator.kyc-exchange-url}")
+    private String kycExchangeUrl;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private KeymanagerService keymanagerService;
+
+    @Autowired
+    private AuthenticationHelperService authenticationHelperService;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @PostConstruct
+    public void initialize() {
+        log.info("Started to setup MOCK IDA");
+    }
+
+    @Validated
+    @Override
+    public KycAuthResult doKycAuth(@NotBlank String relyingPartyId, @NotBlank String clientId,
+                                   @NotNull @Valid KycAuthDto kycAuthDto) throws KycAuthException {
+
+        log.info("Started to build kyc-auth request with transactionId : {} && clientId : {}",
+                kycAuthDto.getTransactionId(), clientId);
+        return authenticationHelperService.doKycAuthImpl(relyingPartyId, clientId, kycAuthDto);
+    }
+
+    @Override
+    public KycExchangeResult doKycExchange(String relyingPartyId, String clientId, KycExchangeDto kycExchangeDto)
```

Comment:
throw exception as not supported for this and below 2 overrided methods

Reply:
done